library/TeslaAuthHelper.cs

```diff
@@ -504,6 +508,51 @@ static string ToHex(byte[] bytes, bool upperCase)
                 result.Append(bytes[i].ToString(upperCase ? "X2" : "x2"));
             return result.ToString();
         }
+
+        public static byte[] GetBytes(String s)
+        {
+            // This is just a passthrough.  We want to make sure that behavior for characters with a
+            // code point value >= 128 is passed through as-is, without depending on your current
+            // machine's default ANSI code page or the exact behavior of ASCIIEncoding.  Some people
+            // are using UTF-8 but that may vary the length of the code verifier, perhaps inappropriately.
+            byte[] bytes = new byte[s.Length];
+            for(int i = 0; i<s.Length; i++)
+                bytes[i] = (byte)s[i];
+            return bytes;
+        }
+
+        public static string Base64UrlEncode(byte[] bytes)
+        {
+            String base64 = Convert.ToBase64String(bytes);
+            String encoded = base64
+                .Replace('+', '-')
+                .Replace('/', '_')
+                .Replace("=", String.Empty)
+                .Trim();

```

Comment:
The `.Trim()` can be removed as the input as AFAIK `Convert.ToBase64String` will never return a white space padded value.